src/main/scala/org/broadinstitute/hail/sparkextras/OrderedLeftJoinRDD.scala

```diff
@@ -9,22 +9,23 @@ import org.broadinstitute.hail.utils.richUtils.RichPairIterator
 import scala.reflect.ClassTag
 
 object OrderedRDDIterator {
-  def apply[PK, K, V](rdd: OrderedRDD[PK, K, V], context: TaskContext, k0: K): OrderedRDDIterator[PK, K, V] = {
-
-    val it = new OrderedRDDIterator(rdd, context)
+  def apply[PK, K, V](rdd: OrderedRDD[PK, K, V], cachedPartitions: Array[Partition],
+    context: TaskContext, k0: K): OrderedRDDIterator[PK, K, V] = {
+    val it = new OrderedRDDIterator(rdd, cachedPartitions, context)
     it.setPartition(rdd.orderedPartitioner.getPartition(k0))
     it
   }
 }
 
 class OrderedRDDIterator[PK, K, V](
   rdd: OrderedRDD[PK, K, V],
+  cachedPartitions: Array[Partition],
```

Comment:
Did you not want to just rename this `partitions`?  Perhaps with a quick comment on why you can't just access `rdd.partitions`?